File: src/main/resources/statics/js/modules/unimall/unimalluser.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'unimall/unimalluser/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '电话', name: 'phone', index: 'phone', width: 80 }, 			
			{ label: '密码', name: 'password', index: 'password', width: 80 }, 			
			{ label: '登陆类型', name: 'loginType', index: 'login_type', width: 80 }, 			
			{ label: '登陆名称', name: 'loginname', index: 'loginname', width: 80 }, 			
			{ label: '邮箱', name: 'email', index: 'email', width: 80 }, 			
			{ label: '', name: 'openId', index: 'open_id', width: 80 }, 			
			{ label: '昵称', name: 'nickname', index: 'nickname', width: 80 }, 			
			{ label: '头像url', name: 'avatarUrl', index: 'avatar_url', width: 80 }, 			
			{ label: '0.普通会员 1.VIP会员', name: 'level', index: 'level', width: 80 }, 			
			{ label: '生日', name: 'birthday', index: 'birthday', width: 80 }, 			
			{ label: '性别', name: 'gender', index: 'gender', width: 80 }, 			
			{ label: '最近登陆时间', name: 'gmtLastLogin', index: 'gmt_last_login', width: 80 }, 			
			{ label: '最近登陆ip', name: 'lastLoginIp', index: 'last_login_ip', width: 80 }, 			
			{ label: '1 正常 2禁用 3注销 ', name: 'status', index: 'status', width: 80 }, 			
			{ label: '', name: 'gmtUpdate', index: 'gmt_update', width: 80 }, 			
			{ label: '', name: 'gmtCreate', index: 'gmt_create', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		unimallUser: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.unimallUser = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.unimallUser.id == null ? "unimall/unimalluser/save" : "unimall/unimalluser/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.unimallUser),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "unimall/unimalluser/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "unimall/unimalluser/info/"+id, function(r){
                vm.unimallUser = r.unimallUser;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});

File: src/main/java/com/icloud/modules/unimall/entity/UnimallUser.java
package com.icloud.modules.unimall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-05-16 10:38:18
 */
@Data
@TableName("t_unimall_user")
public class UnimallUser implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /*  */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /* 电话 */
       @TableField("phone")
       private String phone;
   	   	   /* 密码 */
       @TableField("password")
       private String password;
   	   	   /* 登陆类型 */
       @TableField("login_type")
       private Integer loginType;
   	   	   /* 登陆名称 */
       @TableField("loginname")
       private String loginname;
   	   	   /* 邮箱 */
       @TableField("email")
       private String email;
   	   	   /*  */
       @TableField("open_id")
       private String openId;
   	   	   /* 昵称 */
       @TableField("nickname")
       private String nickname;
   	   	   /* 头像url */
       @TableField("avatar_url")
       private String avatarUrl;
   	   	   /* 0.普通会员 1.VIP会员 */
       @TableField("level")
       private Integer level;
   	   	   /* 生日 */
       @TableField("birthday")
       private Date birthday;
   	   	   /* 性别 */
       @TableField("gender")
       private Integer gender;
   	   	   /* 最近登陆时间 */
       @TableField("gmt_last_login")
       private Date gmtLastLogin;
   	   	   /* 最近登陆ip */
       @TableField("last_login_ip")
       private String lastLoginIp;
   	   	   /* 1 正常 2禁用 3注销  */
       @TableField("status")
       private Integer status;
   	   	   /*  */
       @TableField("gmt_update")
       private Date gmtUpdate;
   	   	   /*  */
       @TableField("gmt_create")
       private Date gmtCreate;
   	
}
